dio/lib/src/adapters/io_adapter.dart

```diff
@@ -227,7 +238,13 @@ class IOHttpClientAdapter implements HttpClientAdapter {
     );
   }
 
-  HttpClient _configHttpClient(Duration? connectionTimeout) {
+  HttpClient _configHttpClient(Duration? connectionTimeout,
+      {bool newClient = false}) {
+    if (newClient) {
+      final client = _createHttpClient();
```

Comment:
If we create new clients here, what about the closing after the client has been used?